File: foundation/foundation-core-java/src/main/java/com/boozallen/aissemble/core/policy/configuration/policymanager/DefaultPolicyManager.java
package com.boozallen.aissemble.core.policy.configuration.policymanager;

/*-
 * #%L
 * Policy-Based Configuration::Policy Manager
 * %%
 * Copyright (C) 2021 Booz Allen
 * %%
 * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
 * #L%
 */

/**
 * {@link DefaultPolicyManager} represents the singleton policy manager that can
 * reuse the methods laid out by the {@link AbstractPolicyManager}.
 * 
 * @author Booz Allen Hamilton
 *
 */
public class DefaultPolicyManager extends AbstractPolicyManager {

    private static DefaultPolicyManager instance = null;

    private DefaultPolicyManager() {
        super();
    }

    public static DefaultPolicyManager getInstance() {
        if (instance == null) {
            synchronized (DefaultPolicyManager.class) {
                instance = new DefaultPolicyManager();
            }
        }
        return instance;
    }

}

File: foundation/aissemble-foundation-core-python/src/policy_manager/default_policy_manager.py
###
# #%L
# Policy-Based Configuration::Policy Manager (Python)
# %%
# Copyright (C) 2021 Booz Allen
# %%
# This software package is licensed under the Booz Allen Public License. All Rights Reserved.
# #L%
###
from .abstract_policy_manager import AbstractPolicyManager


class DefaultPolicyManager(AbstractPolicyManager):
    """
    DefaultPolicyManager represents the singleton policy manager that can
    reuse the methods laid out by the AbstractPolicyManager.
    """

    __instance = None

    def __init__(self):
        if DefaultPolicyManager.__instance is not None:
            raise Exception("Class is a singleton")
        else:
            super().__init__()
            DefaultPolicyManager.__instance = self

    @staticmethod
    def getInstance():
        if DefaultPolicyManager.__instance is None:
            DefaultPolicyManager()
        return DefaultPolicyManager.__instance
